code/modules/cm_aliens/structures/fruit.dm

```diff
@@ -125,7 +125,13 @@
 		add_filter("fruity_glow", 1, list("type" = "outline", "color" = glow_color, "size" = 1))
 	update_icon()
 
-/obj/effect/alien/resin/fruit/proc/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)
+/obj/effect/alien/resin/fruit/proc/can_consume_check(mob/living/carbon/xenomorph/recipient)
+	if(LAZYISIN(recipient.modifier_sources, name))
+		to_chat(recipient, SPAN_XENOWARNING("You're already under the effects of this fruit, go out and kill!"))
+		return
+
+/obj/effect/alien/resin/fruit/proc/consume_effect(mob/living/carbon/xenomorph/recipient, do_consume = TRUE)//why do we bother making this if we're going to overite it later?
+	can_consume_check(recipient)
```

Comment:
This isn't exactly going to do anything, you need to return a boolean value and do an if check.
Then early return if the check does not pass.

Also, the name of your function is a bit verbose, `can_consume` is a good enough name.